### ISSUE TITLE
retire the v3-legacy branch once desired command line functionality is ported to master

### ISSUE THREAD

**Issue by jmartin-sul** [sul-dlss/pre-assembly#221]
not happening in the current work cycle.

**Comment by atz** [comment]
If ever.  That's not a quip, I mean that it might be more sound architecture and security-wise to locate that command-line access somewhere else.

**Comment by blalbrit** [comment]
How about if @peetucket and I put our heads together and make a rational work list for this. I think it's more than a maintenance cycle can handle (though with this infrastructure team, who knows - y'all are blazing through backlogs like nobody's business).

**Comment by jcoyne** [comment]
@blalbrit Yes, I don't think this work will be complete this workcycle, but it would be great if we could get documented what we think is needed.

**Comment by mjgiarlo** [comment]
Sounds good, @blalbrit!

**Comment by blalbrit** [comment]
"desired command line functionality":
@peetucket and I put our heads together and I think that the following features are what we would want ported over into the webapp:

- [ ] ability to set publish/shelve/preserve attributes based on mimetype (see https://github.com/sul-dlss/pre-assembly/issues/440)
- [ ] update list of content structure easily (that is, develop a pattern for adding new models like is being requested in https://github.com/sul-dlss/pre-assembly/issues/427 )
- [ ] possibly beef up worker support for very large projects (on the order of > 100K objects in a pre-assembly run)

We also think it would make sense to split the remediation framework out of pre-assembly into its own project (especially thinking out a year when Fedora is not our back-end and we'll want a framework for large-scale content remediation in addition to the specific tasks supported by Argo). Particularly https://github.com/sul-dlss/pre-assembly/tree/v3-legacy/lib/pre_assembly/remediation - though we can talk about the requirements for this and determine a path forward.
